File: server/CinemasService/Models/Cinema.cs
using Common.Enums;
using Common.Models;
using System.ComponentModel.DataAnnotations;

namespace CinemasService.Models
{
    public class Cinema : AuditableModel
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        [MaxLength(128)]
        public required string Name { get; set; }
        [MaxLength(128)]
        public required string Location { get; set; }
        [MaxLength(512)]
        public required string Description { get; set; }
        public required int Capacity { get; set; }

        [MaxLength(256)]
        public string? ImageUrl { get; set; }
        public EntityStatus Status { get; set; } = EntityStatus.Draft;
        public ICollection<Room> Rooms { get; set; } = new List<Room>();
    }
}

File: ui/cms/src/model/cinema/Cinema.ts
import { EntityStatus } from "../common/EntityStatus";
import { RoomSummaryShort } from "../room/Room";

export interface Cinema {
  id: string;
  name: string;
  location: string;
  description: string;
  capacity: number;
  imageUrl?: string;
  status: EntityStatus;
  createdAt: Date;
  rooms: RoomSummaryShort[];
}

export interface CinemaSummary {
  id: string;
  name: string;
  location: string;
  description: string;
  capacity: number;
  imageUrl?: string;
  status: EntityStatus;
  createdAt: Date;
}

export interface CinemaSummaryShort {
  id: string;
  name: string;
  location: string;
  imageUrl?: string;
  createdAt: Date;
}

export interface CinemaCreateDto {
  name: string;
  location: string;
  description: string;
  capacity: number;
  imageUrl?: string | null | undefined;
}

export interface CinemaUpdateDto extends Partial<CinemaCreateDto> {
  status?: EntityStatus
}
